Lib/test/test_ntpath.py

```diff
@@ -889,6 +890,26 @@ def test_isjunction(self):
                 self.assertFalse(ntpath.isjunction('tmpdir'))
                 self.assertPathEqual(ntpath.realpath('testjunc'), ntpath.realpath('tmpdir'))
 
+    @unittest.skipIf(sys.platform != 'win32', "drive letters are a windows concept")
+    def test_isfile_driveletter(self):
+        current_drive = os.path.splitdrive(os.path.abspath(__file__))[0] + "\\"
+        self.assertFalse(os.path.isfile(current_drive))
```

Comment:
The reason this is false is because a relative drive path like "C:" resolves to the working directory on the drive. Did you want to test a volume device path like `r'\\?\C:'` instead?

Reply:
Ah, yes.  And this is exactly the thing that (erroneously) returns `True` without checking the error code from `GetFileInformationByHandleEx`.